search-service/src/test/kotlin/com/egm/stellio/search/web/TemporalEntityHandlerTests.kt

```diff
@@ -1095,6 +1097,128 @@ class TemporalEntityHandlerTests {
             )
     }
 
+    @Test
+    fun `delete temporal entity should return a 204 if an entity has been successfully deleted`() {
+        coEvery { entityPayloadService.checkEntityExistence(entityUri) } returns Unit.right()
+        coEvery { authorizationService.userCanAdminEntity(entityUri, sub) } returns Unit.right()
+        coEvery { entityPayloadService.deleteEntityPayload(any()) } returns Unit.right()
+        coEvery { authorizationService.removeRightsOnEntity(any()) } returns Unit.right()
+
+        webClient.delete()
+            .uri("/ngsi-ld/v1/temporal/entities/$entityUri")
+            .exchange()
+            .expectStatus().isNoContent
+            .expectBody().isEmpty
+
+        coVerify {
+            entityPayloadService.checkEntityExistence(entityUri)
+            authorizationService.userCanAdminEntity(eq(entityUri), eq(sub))
+            entityPayloadService.deleteEntityPayload(eq(entityUri))
+            authorizationService.removeRightsOnEntity(eq(entityUri))
+        }
+    }
+
+    @Test
+    fun `delete temporal entity should return a 404 if entity to be deleted has not been found`() {
+        coEvery {
+            entityPayloadService.checkEntityExistence(entityUri)
+        } returns ResourceNotFoundException(entityNotFoundMessage(entityUri.toString())).left()
+
+        webClient.delete()
+            .uri("/ngsi-ld/v1/temporal/entities/$entityUri")
+            .exchange()
+            .expectStatus().isNotFound
+            .expectBody().json(
+                """
+                {
+                  "type":"https://uri.etsi.org/ngsi-ld/errors/ResourceNotFound",
+                  "title":"The referred resource has not been found",
+                  "detail":"${entityNotFoundMessage(entityUri.toString())}"
+                }
+                """.trimIndent()
+            )
+    }
+
+    @Test
+    fun `delete temporal entity should return a 404 if entity to be deleted has not been validated`() {
```

Comment:
the status code is 400, not 404. and "entity to be deleted has not been validated" is hard to understand